src/overlays/actors/ovl_En_Zl3/z_en_zl3.c

```diff
@@ -15,421 +19,2827 @@ void EnZl3_Destroy(Actor* thisx, GlobalContext* globalCtx);
+void func_80B591BC(EnZl3* this, GlobalContext* globalCtx);
+void func_80B5922C(EnZl3* this, GlobalContext* globalCtx);
+void func_80B592A8(EnZl3* this, GlobalContext* globalCtx);
+void func_80B59340(EnZl3* this, GlobalContext* globalCtx);
+void func_80B593D0(EnZl3* this, GlobalContext* globalCtx);
+s32 func_80B5944C(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, EnZl3* thisx,
+                  Gfx** gfx);
+void func_80B59AD0(EnZl3* this, GlobalContext* globalCtx);
+void func_80B59DB8(EnZl3* this, GlobalContext* globalCtx);
+void func_80B59FE8(EnZl3* this, GlobalContext* globalCtx);
+void func_80B59FF4(EnZl3* this, GlobalContext* globalCtx);
+void func_80B5A1D0(EnZl3* this, GlobalContext* globalCtx);
+
+static ColliderCylinderInit_Set3 sCylinderInit = {
+    { COLTYPE_UNK0, 0x00, 0x00, 0x09, COLSHAPE_CYLINDER },
+    { 0x00, { 0x00000000, 0x00, 0x00 }, { 0x00000000, 0x00, 0x00 }, 0x00, 0x00, 0x01 },
+    { 25, 80, 0, { 0, 0, 0 } },
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B533B0.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B533FC.s")
-
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/EnZl3_Destroy.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B53468.s")
+UNK_TYPE D_80B5A43C[] = {
+    0x060030C8, 0x06003C48, 0x06004048, 0x06004548, 0x06004948, 0x06004D48, 0x06005148, 0x00000000
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B53488.s")
+UNK_TYPE D_80B5A45C[] = { 0x06003508, 0x06005548, 0x06005948 };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B534CC.s")
+s32 D_80B5A468 = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B53554.s")
+Vec3f D_80B5A46C = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B53568.s")
+Vec3f D_80B5A478 = { 0.0f, 10.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B5357C.s")
+f32 D_80B5A484 = 0.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B53614.s")
+Vec3f D_80B5A488 = { 0.0f, 0.0f, 0.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B5366C.s")
+s32 D_80B5A494 = -1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B536B4.s")
+Vec3f D_80B5A498 = { 148.0f, 260.0f, -87.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B536C4.s")
+Vec3f D_80B5A4A4 = { -12.0f, 260.0f, -147.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B53764.s")
+Vec3f D_80B5A4B0 = { 42.0f, 260.0f, 13.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B537E8.s")
+u32 D_80B5A4BC = 0;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_Zl3/func_80B538B0.s")
+static EnZl3ActionFunc sActionFuncs[] = {
```

Comment:
It might be good to consider moving these function arrays down to where they are used, so you don't have to forward declare all their functions.

Reply:
Ah that cleans it up a bunch! nice